### DISCUSSION TITLE
Filament database Notifications

### DISCUSSION THREAD

**Discussion by JayrousJr** [community/community#63534] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

I am trying to build a dashboard for my system using filament, in the message part of my system I want users to send message even if they are not logged in. I therefore want to display the notifications in my filament dashboard showing to me that a message has been sent to the website as well as I want to get the on resource notification. The message is being worked on in the MessagesController, how can I implement this. I need your ideas fellow coders

**Comment by rifqifarhansyah** [comment]
It sounds like you're working on building a dashboard using Filament and you want to implement a feature where users can send messages even if they're not logged in, and you want to display notifications for those messages on your dashboard. Additionally, you want to receive notifications when a message is sent to your website. To achieve this, you'll need to set up a combination of frontend and backend components. Here's a general outline of how you could approach this:

1:Frontend Implementation:
-Create a message form in your frontend interface that allows users to enter messages and submit them.
-When a user submits a message, send an API request to your backend to store the message in your database.
2.Backend Implementation:
-Create an API endpoint, such as /api/messages, that accepts incoming messages from the frontend.
-In your MessagesController, handle the incoming messages by storing them in your database (e.g., in a table named messages).
-If the user is not logged in, you can still capture their name and email along with the message. You could allow them to send messages without requiring authentication, but you would need to be cautious about potential misuse or spam.
3.Notification System:
-Implement a notification system that alerts you when a new message is received. You could use email notifications, push notifications, or any other preferred method.
-In your backend, after saving the message, trigger the notification system to send you a notification about the new message.
4.Displaying Notifications in Filament:
-In your Filament dashboard, you can create a section to display notifications.
-To show notifications for new messages, you would need to query your database for unread messages and display them in your Filament dashboard.
5.Marking Messages as Read:
-When you view a message in your Filament dashboard, you should mark it as read to avoid showing the same notification again.
-Create an API endpoint to update the status of a message to "read" once it's viewed.
-Remember, the implementation details can vary based on the technologies you're using, the structure of your project, and your specific requirements. It's important to design the system with security in mind, especially if you're allowing messages from anonymous users.

If you're using specific frameworks or technologies, it might be helpful to refer to their documentation or seek help from their respective communities to get more detailed assistance.
